modules/cloudtrail/main.tf

```diff
@@ -0,0 +1,328 @@
+data "aws_caller_identity" "current" {}
+
+resource "random_id" "bucket_name" {
+  byte_length = 16
+}
+
+resource "aws_cloudtrail" "ksoc_discovery" {
+  name                          = var.resources_prefix
+  s3_bucket_name                = aws_s3_bucket.ksoc_discovery.id
+  include_global_service_events = true
+  is_multi_region_trail         = true
+  kms_key_id                    = aws_kms_key.ksoc_discovery_key.arn
+  event_selector {
+    include_management_events        = "true"
+    read_write_type                  = "WriteOnly"
+    exclude_management_event_sources = ["kms.amazonaws.com", "rdsdata.amazonaws.com"]
+  }
+  sns_topic_name = aws_sns_topic.ksoc_discovery.name
+}
+
+resource "aws_s3_bucket" "ksoc_discovery" {
+  bucket        = "${var.resources_prefix}-${random_id.bucket_name.hex}"
```

Comment:
I think these buckets should named `ksoc-connect-cloudtrail-<uuid>`

Reply:
It is called `ksoc-discovery-<random_hash>`, not sure if adding `cloudtrail` is that helpful, but can add it.